### DISCUSSION TITLE
is it easier to fork firefox?

### DISCUSSION THREAD

**Discussion by al-caveman** [qutebrowser/qutebrowser#6127] [Ideas] [answered]
easier than what?  easier than granting qutebrowser a good ad blocker (like uBlock Origin), fingerprinting resistance (like firefox), gpu accelerated video, etc.

[here](https://github.com/tridactyl/godmode) is some work for giving firefox god-mode to let `tridactyl` work in every page.  but still incomplete.  

currently, tridactyle has limitations compared to qutebrowser (e.g. doesn't run on all pages), but that's all about it, as everything else seem to be an advantage:  

- firefox has  all those important features above (good ad blocker, fingerprinting resistance, gpu acceleration, etc), 
- some unexpected advantages, such as tridactyle's hinting is attached to the page, so scrolling around will cause hints to scroll with the site (with qb, the hints are attached to the window, irrespective of the scrolling), and hinting also searches alt image texts (qb's hinting doesn't search alt texts of imags).

so i wonder:  would it be easier to just maintain a firefox patch that makes tridactyle complete?  e.g. complete the `godmode` patch, enhance tridactyl, and perhaps also add extra patches to make the tabs/scrollbars/etc more `vim`y?  the theme change is perhaps as easy as patching a css file.  distros can then distribute a package `firefox-vimed`.

**Comment by The-Compiler** [comment, marked as answer]
Browsers like Firefox or Chromium move at an [incredible speed](https://hg.mozilla.org/mozilla-central/shortlog) - we're talking hundreds of changes every day, and often millions of changed lines for every release.

Keeping up with that in a fork (even more so in one which does deeper changes than just removing a couple of extension limits) is not realistic without a handful of full-time (or at least part-time) developers. There are a couple of projects pulling it off (think Ungoogled Chromium or maybe Pale Moon), but it's not something I'm interested in doing.
